overrides/runner.rb

```diff
@@ -23,76 +23,122 @@ module Overrides
   # description:
   #
   # overrides: !ruby/object:Overrides::ResourceOverrides
+  #   product: # Allows us to override information about the product.
+  #     prefix: magicproduct
```

Comment:
Is this an actually override we have / have we overriden the product for a resource yet? I can see it being a useful feature, I'm just not familiar with any occurences. Also I'd lean towards `legacy_name` over `prefix`, similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/master/products/cloudbuild/terraform.yaml#L15